### PR TITLE
add a README for build_web_compilers

### PR DESCRIPTION
**Pull request by jakemac53**
Fixes https://github.com/dart-lang/build/issues/847

### CODE REVIEW

File: build_web_compilers/README.md [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+following to your `pubspec.yaml`:
+
+```yaml
+dev_dependencies:
+  build_web_compilers:
+```
+
+## Usage
+
+If you are using the autogenerated build script (going through `pub run` instead
```
Comment (natebosch) [inline build_web_compilers/README.md]: Maybe `pub run build_runner` so users don't get confused with other `pub run` things

File: build_web_compilers/README.md [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+of handwriting a `build.dart` file), then all you need is the `dev_dependency`
+listed above.
+
+If you are using a custom build script, you will need to add the following
```
Comment (natebosch) [inline build_web_compilers/README.md]: I wonder if we should expose a `List<BuilderApplication>` for this use case?

Comment (natebosch) [inline build_web_compilers/README.md]: Call out that these must be last.

Also, should we consider moving this below configuration to an 'advanced usage' or 'manual usage' section?

File: build_web_compilers/README.md (hunk 1 above)
Comment (jakemac53) [reply build_web_compilers/README.md]: done

File: build_web_compilers/README.md (hunk 2 above)
Comment (jakemac53) [reply build_web_compilers/README.md]: we could consider doing that.... it would likely be more future proof

Comment (jakemac53) [reply build_web_compilers/README.md]: Moved it under a `Manual Usage` section

**Pull request merged**